File: backend/app/Models/Announcement.php
<?php

namespace App\Models;

use App\Models\Concerns\BelongsToInstitution;
use Database\Factories\AnnouncementFactory;
use Illuminate\Database\Eloquent\Attributes\Fillable;
use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

#[Fillable(['institution_id', 'title', 'body', 'status', 'published_at', 'created_by'])]
class Announcement extends Model
{
    /** @use HasFactory<AnnouncementFactory> */
    use BelongsToInstitution, HasFactory, HasUuids, SoftDeletes;

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'status' => 'string',
            'published_at' => 'datetime',
        ];
    }

    /**
     * User who created the announcement.
     */
    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by');
    }
}

File: mobile/lib/models/announcement.dart
class Announcement {
  final String id;
  final String institutionId;
  final String title;
  final String body;
  final String status;
  final String? publishedAt;
  final String? createdAt;

  const Announcement({
    required this.id,
    required this.institutionId,
    required this.title,
    required this.body,
    required this.status,
    this.publishedAt,
    this.createdAt,
  });

  factory Announcement.fromJson(Map<String, dynamic> json) {
    return Announcement(
      id: json['id'] as String? ?? '',
      institutionId: json['institution_id'] as String? ?? '',
      title: json['title'] as String? ?? '',
      body: json['body'] as String? ?? '',
      status: json['status'] as String? ?? 'published',
      publishedAt: json['published_at'] as String?,
      createdAt: json['created_at'] as String?,
    );
  }
}
